### DISCUSSION TITLE
IE breaking with syntax error

### DISCUSSION THREAD

**Discussion by vinodhum** [react-hook-form/react-hook-form#2603] [Q&A] [answered]
Hi,
react-hook-form is throwing error on IE11 with "var isHTMLElement = (value) => value instanceof HTMLElement;"
is react-hook-form supported on IE ?
any idea how to make it work on IE, I really love react-hook-form but if it not supported then i have re-code everything again.

appreciate your help.
Thank You

**Comment by vinodhum** [comment, marked as answer]
I was abel to make it working using faq with https://react-hook-form.com/faqs
it says for IE we have to import { useForm } from 'react-hook-form/dist/index.ie11'; // V6
how can we import based on browser if its IE or Chrome ?

**Reply by bluebill1049** [reply]
we can't, you have to use either :(

**Comment by redshoga** [comment]
I solved this problem by setting up the project's transpile configuration.

Here's an example of a Next.js project. 

```js
const withTM = require('next-transpile-modules')(['react-hook-form']);

module.exports = withTM(
  // webpack config
  // ...
);
```

**Reply by bluebill1049** [reply]
thanks for sharing solutions @redshoga 🙏

**Reply by c0d3x** [reply]
I am having same issue when using resolver, I get same problem in IE11 with the IE11 version.

import { useForm, Controller } from 'react-hook-form/dist/index.ie11';
import { yupResolver } from '@hookform/resolvers/dist/ie11/yup';

**Comment by MkMan** [comment]
Using the IE version from `dist` worked for me BUT I initially wasn't using it for other parts of RHF (e.g. `Controller`) and it took me a day to debug 😭 . So, for anyone who has this issue, make sure **all** your imports are from the IE file in `dist` 😃
